index.js

```diff
@@ -231,20 +237,43 @@ Client.prototype.config = function (opts) {
   // fixes bug where cached/memoized _encodedMetadata wouldn't be
   // updated when client was reconfigured
   if (this._encodedMetadata) {
-    this.updateEncodedMetadata()
+    this._resetEncodedMetadata()
```

Comment:
Is it worth mentioning in the docs that filters will be applied _whenever_ the agent is reconfigured? (not just on startup)?

Reply:
Yup, that is worth it. Added in commit 9a12928